amd_openvx_extensions/amd_nn/src/topk_layer.cpp

```diff
@@ -0,0 +1,270 @@
+        data->axis = data->axis - count;
+    }
+
+    //vector to sort indices
+    std::vector<size_t> idx(input_dims_0[data->axis]);  
+    //temporary moving ptr
+    float *x_tensor_temp = data->x_tensor_buffer;    
+    
+    for(int i = 0; i < count_input_dims_0; i += input_dims_0[data->axis])
+    {
+        std::iota(idx.begin(), idx.end(), 0); 
+        if (data->largest)
+        {   
+            std::sort(idx.begin(), idx.end(), [&x_tensor_temp](const size_t &a, const size_t &b)
+                                                { return x_tensor_temp[a] > x_tensor_temp[b];});   
+        }
+        else
+        { 
+            std::sort(idx.begin(), idx.end(), [&x_tensor_temp](const size_t &a, const size_t &b)
+                                               { return x_tensor_temp[a] < x_tensor_temp[b];});  
+        }
+        
+        //keep only top k elements
+        int keep_elements = (input_dims_0[data->axis] < data->k_tensor_buffer[0]) ? input_dims_0[data->axis]:data->k_tensor_buffer[0];
+        for (int j = 0; j < keep_elements; j++)
+        {
+            values.push_back(x_tensor_temp[idx[j]]);
+            indices.push_back(idx[j]);
+        }
+        x_tensor_temp += input_dims_0[data->axis];
+    }
+
+    float* values_ptr = &values[0];
+    int64_t* indices_ptr = &indices[0];
+
+    //finding size of topk output and assigning stride
+    output_dims_0[3] = 1; 
+    output_dims_0[2] = 1;
+    output_dims_0[1] = 1; 
+    output_dims_0[0] = values.size(); //total count of values. Acts as a 1D array
+
+    output_dims_1[3] = 1; 
+    output_dims_1[2] = 1;
+    output_dims_1[1] = 1; 
+    output_dims_1[0] = indices.size(); //total count of indices. Acts as a 1D array    
+
+    vx_size stride_output_0[4] = {sizeof(float), output_dims_0[0]*sizeof(float), output_dims_0[0]*output_dims_0[1]*sizeof(float), output_dims_0[0]*output_dims_0[1]*output_dims_0[2]*sizeof(float) };
+    vx_size stride_output_1[4] = {sizeof(int64_t), output_dims_1[0]*sizeof(int64_t), output_dims_1[0]*output_dims_1[1]*sizeof(int64_t), output_dims_1[0]*output_dims_1[1]*output_dims_1[2]*sizeof(int64_t) };
+
+    ERROR_CHECK_STATUS(vxCopyTensorPatch((vx_tensor)parameters[5], 4, nullptr, nullptr, stride_output_0, values_ptr, VX_WRITE_ONLY, VX_MEMORY_TYPE_HOST));
+    ERROR_CHECK_STATUS(vxCopyTensorPatch((vx_tensor)parameters[6], 4, nullptr, nullptr, stride_output_1, indices_ptr, VX_WRITE_ONLY, VX_MEMORY_TYPE_HOST));
+
+    return VX_SUCCESS;
+}
+
+static vx_status VX_CALLBACK initializeTopK(vx_node node, const vx_reference *parameters, vx_uint32 num)
+{
+	data = new TopKLayerLocalData;
+	memset(data, 0, sizeof(*data));
+
+	vx_size input_dims_0[4];
+    vx_size count_input_dims_0 = input_dims_0[0]*input_dims_0[1]*input_dims_0[2]*input_dims_0[3];
+
+    ERROR_CHECK_STATUS(vxQueryTensor((vx_tensor)parameters[0], VX_TENSOR_DIMS, input_dims_0, sizeof(input_dims_0)));
+
+    data->x_tensor_buffer = (float*)malloc(count_input_dims_0*sizeof(float));
+

```

Comment:
Looks like vxSetNodeAttribute(node, VX_NODE_LOCAL_DATA_PTR, ..) is not called for local data